ADR/0033-upstreaming-integration-service.md

```diff
@@ -0,0 +1,228 @@
+identify which [IntegrationTestScenarios] it is relevant to.
+
+On the [IntegrationTestScenario], drop the `application` from the spec and replace it with
+a [selector] which the user can use to reference the `LABELNAME` label on [TestSubjects]. It should
+be possible for the user to specify a selector that references any attribute they wish. The user
+should conclude naturally that `LABELNAME` is a good candidate for usage in their selectors. UI and
+docs can guide this.
+
+- A new [TestSubject] without a `LABELNAME` label should be mutated by webhook to gain a value of
+  "default".
+- Attempts to remove the `LABELNAME` label from a [TestSubject] should be rejected (or mutated to
+  "default"?)
+- Attempts to edit the `LABELNAME` label on a [TestSubject] should be permitted.
+
+### The Global Candidate List becomes the control TestSubject, the last TestSubject to pass testing
+
+Today, the "global candidate list" is a term that we use to refer to the list of all the built-image
+pullspecs attached to every [Component] of an [Application]. Replace that implementation
+with an identifying *label* on a [TestSubject] in the namespace: `CONTROLLABEL`. Call the [TestSubject] that
+carries this `CONTROLLABEL` label the **control** TestSubject for the `LABELNAME` label.
+
+[integration-service] should grow a mutating webhook that *removes* the `CONTROLLABEL` label from
+the last control [TestSubject] for the `LABELNAME` label, when the `CONTROLLABEL` label is applied
+to a new [TestSubject] bearing the `LABELNAME` label.
+
+### TestSubject Construction as optional behavior
+
+**Test Initiation** has a clear control plane: the [IntegrationTestScenario] combined with the
+[TestSubject]. Meaning, when the user creates an [IntegrationTestScenario] and a [TestSubject], and
+the [selector] of the [IntegrationTestScenario] matches the attributes of [TestSubject], then together
+they represent intent for the subject to be tested. The user uses these resources to control what
+the system does.
+
+Today, control of the **Snapshot Construction** process is mixed up in the [Application] and [Component]
+model, and is partly just implied. We *always* construct [Snapshots].
+
+**TestSubject Construction** (called Snapshot Construction in our system today) should have a clear
+control plane resource: the [ToBeDetermined] resource.
+
+TODO - describe this in more detail
+
+TODO - describe the spec of this resource.
+
+### Release Initiation, as owned by the release-service
+
+Note, in this pattern, [Release] construction looks a lot like [TestSubject] construction. The
+triggered system knows enough about the triggering system to optionally initiate itself.
+
+TODO - describe this in more detail
+
+> :memo: Another alternative could be to establish an overarching workflow operator that knows how
+> to bridge the gaps (create TestSubjects and create Releases). TODO what would this look like? Too
+> much domain knowledge in one place?
+
+## Use Cases in Detail
```

Comment:
As I read through these use cases, modifying integration service into an event bus keeps coming to mind. Are there efficiencies that we can gain or use cases that we can improve if we move away from managing state solely via CRs? Triggering from external locations _seems_ more intuitive with a bus.
Would there be any critical functionality/data that we lose if we move to a bus?

It would be a new use case for the integration service and I am not sure where/how it would fit in best, but what if the integration service attested to the artifacts with the result of each ITS's run? It would likely not be an _easy_ solution today (potentially easier with the referrer's API), but would it be any easier "tomorrow" with one of the integration services's APIs over the other?